Application tests failing w/ Quarkus 3.13.0 + `io.smallrye.config:smallrye-config-source-keystore`

### Describe the bug

All `@QuarkusTest` and `@QuarkusIntegrationTest` fail with an NPE at `SmallRyeConfig.java:751`.

Quarkus 3.12.3 worked fine.

### Expected behavior

_No response_

### Actual behavior

Full stack trace
```
java.lang.NullPointerException: Cannot read field "interceptorChain" because "this.configSources" is null
java.lang.RuntimeException: java.lang.NullPointerException: Cannot read field "interceptorChain" because "this.configSources" is null
	at io.quarkus.test.junit.QuarkusTestExtension.throwBootFailureException(QuarkusTestExtension.java:634)
	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestClassConstructor(QuarkusTestExtension.java:718)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: java.lang.NullPointerException: Cannot read field "interceptorChain" because "this.configSources" is null
	at io.smallrye.config.SmallRyeConfig.interceptorChain(SmallRyeConfig.java:751)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.restart(SmallRyeConfigSourceInterceptorContext.java:33)
	at io.quarkus.deployment.configuration.ConfigCompatibility.quarkusPackageJarManifestAddImplementationEntries(ConfigCompatibility.java:494)
	at io.quarkus.deployment.configuration.ConfigCompatibility$BackEnd.getValue(ConfigCompatibility.java:211)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:25)
	at io.smallrye.config.SmallRyeConfigSources.getValue(SmallRyeConfigSources.java:41)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:25)
	at io.smallrye.config.SecretKeysConfigSourceInterceptor.getValue(SecretKeysConfigSourceInterceptor.java:26)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:25)
	at io.smallrye.config.RelocateConfigSourceInterceptor.getValue(RelocateConfigSourceInterceptor.java:25)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:25)
	at io.smallrye.config.RelocateConfigSourceInterceptor.getValue(RelocateConfigSourceInterceptor.java:25)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:25)
	at io.smallrye.config.RelocateConfigSourceInterceptor.getValue(RelocateConfigSourceInterceptor.java:25)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:25)
	at io.smallrye.config.ProfileConfigSourceInterceptor.getValue(ProfileConfigSourceInterceptor.java:46)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:25)
	at io.smallrye.config.LoggingConfigSourceInterceptor.getValue(LoggingConfigSourceInterceptor.java:26)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:25)
	at io.smallrye.config.LoggingConfigSourceInterceptor.getValue(LoggingConfigSourceInterceptor.java:26)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:25)
	at org.projectnessie.quarkus.config.datasource.JdbcUrlInterceptor.getValue(JdbcUrlInterceptor.java:36)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:25)
	at io.smallrye.config.RelocateConfigSourceInterceptor.getValue(RelocateConfigSourceInterceptor.java:25)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:25)
	at io.smallrye.config.RelocateConfigSourceInterceptor.getValue(RelocateConfigSourceInterceptor.java:25)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:25)
	at io.smallrye.config.RelocateConfigSourceInterceptor.getValue(RelocateConfigSourceInterceptor.java:25)
	at io.quarkus.smallrye.openapi.runtime.OpenApiConfigMapping.getValue(OpenApiConfigMapping.java:32)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:25)
	at io.smallrye.config.ExpressionConfigSourceInterceptor.getValue(ExpressionConfigSourceInterceptor.java:53)
	at io.smallrye.config.ExpressionConfigSourceInterceptor.getValue(ExpressionConfigSourceInterceptor.java:45)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:25)
	at io.quarkus.opentelemetry.runtime.config.HierarchicalOTelConnectionConfigInterceptor.iterateNames(HierarchicalOTelConnectionConfigInterceptor.java:56)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:44)
	at io.smallrye.config.ConfigSourceInterceptor.iterateNames(ConfigSourceInterceptor.java:55)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:44)
	at io.quarkus.runtime.configuration.QuarkusConfigBuilderCustomizer$3$1.iterateNames(QuarkusConfigBuilderCustomizer.java:101)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:44)
	at io.smallrye.config.AbstractMappingConfigSourceInterceptor.iterateNames(AbstractMappingConfigSourceInterceptor.java:30)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:44)
	at io.smallrye.config.ConfigSourceInterceptor.iterateNames(ConfigSourceInterceptor.java:55)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:44)
	at io.smallrye.config.ConfigSourceInterceptor.iterateNames(ConfigSourceInterceptor.java:55)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:44)
	at io.smallrye.config.ConfigSourceInterceptor.iterateNames(ConfigSourceInterceptor.java:55)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:44)
	at io.quarkus.deployment.configuration.ConfigCompatibility$FrontEnd.iterateNames(ConfigCompatibility.java:132)
	at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:44)
	at io.smallrye.config.SmallRyeConfig$SmallRyeConfigSourceContext.iterateNames(SmallRyeConfig.java:1091)
	at io.smallrye.config.ConfigSourceContext$ConfigSourceContextConfigSource.getPropertyNames(ConfigSourceContext.java:36)
	at io.smallrye.config.SmallRyeConfig.matchPropertiesWithEnv(SmallRyeConfig.java:148)
	at io.smallrye.config.SmallRyeConfig.<init>(SmallRyeConfig.java:91)
	at io.smallrye.config.SmallRyeConfigBuilder.build(SmallRyeConfigBuilder.java:736)
	at io.smallrye.config.source.keystore.KeyStoreConfigSourceFactory.getKeyStoreConfig(KeyStoreConfigSourceFactory.java:90)
	at io.smallrye.config.source.keystore.KeyStoreConfigSourceFactory.getConfigSources(KeyStoreConfigSourceFactory.java:38)
	at io.smallrye.config.ConfigurableConfigSource.getConfigSources(ConfigurableConfigSource.java:50)
	at io.smallrye.config.SmallRyeConfig$ConfigSources.mapLateSources(SmallRyeConfig.java:935)
	at io.smallrye.config.SmallRyeConfig$ConfigSources.<init>(SmallRyeConfig.java:803)
	at io.smallrye.config.SmallRyeConfig.<init>(SmallRyeConfig.java:85)
	at io.smallrye.config.SmallRyeConfigBuilder.build(SmallRyeConfigBuilder.java:736)
	at io.quarkus.deployment.configuration.BuildTimeConfigurationReader.initConfiguration(BuildTimeConfigurationReader.java:410)
	at io.quarkus.deployment.ExtensionLoader.loadStepsFrom(ExtensionLoader.java:137)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:106)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:331)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:254)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:58)
	at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:219)
	at io.quarkus.test.junit.QuarkusTestExtension.ensureStarted(QuarkusTestExtension.java:601)
	at io.quarkus.test.junit.QuarkusTestExtension.beforeAll(QuarkusTestExtension.java:651)
	... 1 more
```

### How to Reproduce?

1. `git clone https://github.com/projectnessie/nessie.git`
2. `git checkout renovate/quarkusplatform`
3. `./gradlew :nessie-quarkus:test --tests TestBasicOperations`


### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

21

### Quarkus version or git rev

4.13.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 8.9

### Additional information

_No response_

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -51,7 +51,7 @@
         <microprofile-lra.version>2.0</microprofile-lra.version>
         <microprofile-openapi.version>3.1.1</microprofile-openapi.version>
         <smallrye-common.version>2.5.0</smallrye-common.version>
-        <smallrye-config.version>3.9.0</smallrye-config.version>
+        <smallrye-config.version>3.9.1</smallrye-config.version>
         <smallrye-health.version>4.1.0</smallrye-health.version>
         <smallrye-metrics.version>4.0.0</smallrye-metrics.version>
         <smallrye-open-api.version>3.10.0</smallrye-open-api.version>

